src/Shared/Data.php

```diff
@@ -199,6 +199,9 @@ public function getPayment($payment_id, $apiKey, $use_cache = true): ?\Mollie\Ap
      */
     public function getAllPaymentMethods($apiKey, $test_mode = false, $use_cache = true)
     {
+        if (!$apiKey) {
+            $apiKey = $this->getApiKey($test_mode);
+        }
```

Comment:
If this block is needed, we probably have a bug somewhere. Every time we call a method `getAllPaymentMethods` we request an API key before the call.

Reply:
yeah, logic not needed I removed it